File: Dev/C++/Mascaret/src/BEHAVE/AgentBehaviorExecution.cpp
#include "BEHAVE/AgentBehaviorExecution.h"

namespace BEHAVE
{

AgentBehaviorExecution::AgentBehaviorExecution(shared_ptr<Behavior> specif,shared_ptr<InstanceSpecification> host, const Parameters& p) : BehaviorExecution(specif,host,p)
{}

AgentBehaviorExecution::~AgentBehaviorExecution()
{}

/*
double AgentBehaviorExecution::execute()
{
   action();
   if (!done()) return 0.2;
   else return 0;
}
 */

}

File: Dev/CS/Mascaret/Mascaret/BEHAVE/AgentBehaviorExecution.cs
using System;
using System.Collections.Generic;

namespace Mascaret
{
    public abstract class AgentBehaviorExecution : BehaviorExecution
    {
        public AgentBehaviorExecution(Behavior specif, InstanceSpecification host, Dictionary<String, ValueSpecification> p)
            : base(specif, host, p, false)
        {
        }

        public AgentBehaviorExecution()
        {

        }

        public abstract void action();

        public abstract bool done();


    }
}

